src/routes/users.ts

```diff
@@ -1,48 +1,72 @@
-import express, { Request, Response } from 'express';
-import { CreateUserParams } from '../interfaces/index';
-import { getUsers, createUser, getUserById, deleteUser, updateUser } from '../controllers/users';
+import express, { NextFunction, Request, Response } from 'express';
+import {
+  CreateUserParams,
+  createUserParams,
+  getUserParams,
+  loginUserParams,
+  LoginUserParams,
+  updateUserParams,
+} from '../interfaces/index';
+import { getUsers, register, getUserById, deleteUser, updateUser, login } from '../controllers/users';
 
 const router = express.Router();
 
 router.get('/', async (req: Request, res: Response) => {
   try {
-    const data = await getUsers();
-    res.send(data);
+    const allUsers = await getUsers();
+    res.send(allUsers);
   } catch (error: any) {
     res.send({ error: error.message });
   }
 });
 
-router.get('/:id', async ({ body }: { body: CreateUserParams }, res: Response) => {
+router.get('/:id', async ({ params }: Request, res: Response) => {
   try {
-    const data = await getUserById(body.id);
-    res.send(data);
+    const { id } = getUserParams.parse(params);
+    const user = await getUserById(id);
+    res.send(user);
   } catch (error: any) {
     res.send({ error: error.message });
   }
 });
 
```

Comment:
We should be calling next() inside the catch and let the error handling middleware deal with the error